### PR TITLE
Add an option to set or add when using AssembleEA.

### PR DESCRIPTION
**Pull request by YohannDudouit**
This PR adds a `const bool add` to the bilininteg `AssembleEA` method. This allows to set the element matrices instead of always adding to them, therefore preventing some unnecessary data movements if there is only one `Integrator`.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1691](https://github.com/mfem/mfem/pull/1691) | @YohannDudouit | @tzanio | @v-dobrev + @tzanio | 08/13/20 | 08/20/20 | 08/25/20 | |

### CODE REVIEW

**Comment by YohannDudouit** [comment]
@v-dobrev @artv3 I added the feature we discussed while debugging the other day.

@artv3 now you can use `AssembleEA(ea_data, false)` if you want to directly set the element matrices.

File: fem/bilininteg.hpp [hunk 1]
```diff
@@ -79,20 +79,23 @@ class BilinearFormIntegrator : public NonlinearFormIntegrator
    virtual void AddMultTransposePA(const Vector &x, Vector &y) const;
 
    /// Method defining element assembly.
-   /** The result of the element assembly is added and stored in the @a emat
-       Vector. */
-   virtual void AssembleEA(const FiniteElementSpace &fes, Vector &emat);
+   /** The result of the element assembly is added to the @a emat Vector if
+       @a add is true, it sets @a emat if @a add is false. */
```
Comment (tzanio) [inline fem/bilininteg.hpp:83]:
```suggestion
       @a add is true. Otherwise, if @a add is false, we set @a emat. */
```

**Comment by v-dobrev** [comment]
LGTM

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**